src/python/gudhi/tensorflow/lower_star_simplex_tree_layer.py

```diff
@@ -0,0 +1,77 @@
+import numpy               as np
+import tensorflow          as tf
+
+#########################################
+# Lower star filtration on simplex tree #
+#########################################
+
+# The parameters of the model are the vertex function values of the simplex tree.
+
+def _LowerStarSimplexTree(simplextree, filtration, dimension):
+    # Parameters: simplextree (simplex tree on which to compute persistence)
+    #             filtration (function values on the vertices of st),
+    #             dimension (homology dimension),
+    
+    for s,_ in simplextree.get_filtration():
+        simplextree.assign_filtration(s, -1e10)
+
+    # Assign new filtration values
+    for i in range(simplextree.num_vertices()):
+        simplextree.assign_filtration([i], filtration[i])
+    simplextree.make_filtration_non_decreasing()
+    
+    # Compute persistence diagram
+    dgm = simplextree.persistence()
+    
+    # Get vertex pairs for optimization. First, get all simplex pairs
+    pairs = simplextree.persistence_pairs()
+    
+    # Then, loop over all simplex pairs
+    indices, pers = [], []
+    for s1, s2 in pairs:
+        # Select pairs with good homological dimension and finite lifetime
```

Comment:
Is it never useful to optimize the birth time of infinite bars?

Reply:
It probably is, but forgetting the infinite points makes life with tensorflow easier as the output shape is fixed.

Reply:
By fixed you mean always of the form `(?,2)` while the infinite features would give something 1-dimensional? We wouldn't connect them to the same later layers anyway. This seems related to the question about several dimensions.

Reply:
You are right. Again, it was easy to modify so now the code outputs both finite and essential features.